test/sqllogictest/transform/is_null_propagation.slt

```diff
@@ -61,28 +61,38 @@ Explained Query:
               Project (#0) // { arity: 1 }
                 Filter (#0 = bigint_to_double(#1)) // { arity: 2 }
                   Get l2 // { arity: 2 }
-              Negate // { arity: 1 }
-                Project (#0) // { arity: 1 }
-                  Filter (#0 = 0) // { arity: 2 }
-                    Get l2 // { arity: 2 }
-              Filter (#0 = 0) // { arity: 1 }
-                Get l1 // { arity: 1 }
+              CrossJoin type=differential // { arity: 1 }
```

Comment:
This regression seems to be the main one to sort out now (https://github.com/MaterializeInc/materialize/pull/25551 cleaned up some others). To be honest, this whole query looks pretty badly planned at the moment (two uses of a cross join, where we could push down filters and turn in to two distinct equijoins). @ggevay and @aalexandrov: any thoughts about what the right standard here should be (this is now an example of "I have no clue why we should be allowed to remove the join", and would instead proceed to try and un-tangle the whole query).

Reply:
I'm pretty sure this is a thing redundant join elimination should be able to handle (the second, new argument to the `CrossJoin` has the same domain as the first, has a unique key of `[]`, etc. However, the "same domain" is "the subset of `l1` equal to `0`" and I can imagine this is easier to reason about without the filters pushed down. By the time that happens, and the following `Project ()`, there's not a lot to go on. I'll investigate join elimination today, though.